constance/backends/database/__init__.py

```diff
@@ -82,19 +87,32 @@ def get(self, key):
 
     def set(self, key, value):
         key = self.add_prefix(key)
+        created = False
+        queryset = self._model._default_manager.all()
+        # Set _for_write attribute as get_or_create method does
+        # https://github.com/django/django/blob/2.2.11/django/db/models/query.py#L536
+        queryset._for_write = True
 
         try:
-            constance = self._model._default_manager.get(key=key)
+            constance = queryset.get(key=key)
         except (OperationalError, ProgrammingError):
             # database is not created, noop
             return
         except self._model.DoesNotExist:
-            old_value = None
-            constance = self._model._default_manager.create(key=key, value=value)
-        else:
+            try:
+                with transaction.atomic(using=queryset.db):
+                    queryset.create(key=key, value=value)
+                created = True
+            except IntegrityError as error:
+                # Allow concurrent writes
+                constance = queryset.get(key=key)
+
+        if not created:
             old_value = constance.value
```

Comment:
**_From PyCharm:_** Local variable `constance` might be referenced before assignment